src/webgpu/shader/values.ts

```diff
@@ -63,3 +63,29 @@ export function normalF32Examples(): Array<number> {
   );
   return result;
 }
+
+/** Returns an array of 32-bit NaNs, as Uint32 bit patterns.
+ * NaNs have: maximum exponent, but the mantissa is not zero.
+ */
+export function nanF32BitsExamples(): Array<number> {
+  const result: number[] = [];
+  const exponent_bit = 0x7f80_0000;
+  const sign_bits: [number, number] = [0, 0x8000_0000];
+  for (const sign_bit in sign_bits) {
```

Comment:
I think you want `for (const sign_bit of sign_bits) {`

`x in y` [gives you the index of each element](https://www.tutorialsteacher.com/typescript/for-loop), and my IDE claims the type of `sign_bit` is a `string` (?!). I'm guessing this is why on L75 you're doing weird casting.

I see you're doing the same thing above, which I think might be buggy. Sorry I didn't spot this before.

Reply:
Yikes. You're right.  Yeah, I was seeing that 'strings' stuff too.

> The for...in statement iterates over all enumerable properties of an object that are keyed by strings ....